source/puppet/4.4/reference/static_catalogs.md

```diff
@@ -0,0 +1,132 @@
+
+### What's a static catalog?
+
+Unlike a standard Puppet catalog, a static catalog includes metadata that identifies the desired state of any [file resources][] on a node that have `source` attributes pointing to `puppet:///` locations. This metadata can refer to a specific version of the file, rather than the latest version, and confirm that the agent is applying the appropriate version of the file resource for the catalog. Also, because much of the metadata is provided in the catalog, Puppet agents make fewer requests to the master.
+
+### Why use static catalogs?
+
+When a Puppet master compiles a normal catalog, the catalog doesn't specify a particular version of its file resources. When the agent applies the catalog, it always retrieves the latest version of that file resource, or uses a previously retrieved version if it matches the latest version's contents.
+
+When a Puppet manifest depends on a file whose contents might change more frequently than the Puppet agent receives new catalogs --- for instance, if the agent is caching catalogs or can't reach a Puppet master over the network --- this can lead to a node applying a version of the referenced file that doesn't match the instructions in the catalog.
+
+Consequently, the agent's Puppet runs might produce different results each time the agent applies the same catalog. This often causes problems, because Puppet generally expects a catalog to produce the same results each time it's applied, regardless of any code or file content updates on the master.
+
+Additionally, each time a Puppet agent applies a normal cached catalog that contains file resources sourced from `puppet:///` locations, the agent requests [file metadata][] from the master each time the catalog's applied, even though nothing's changed in the cached catalog. This causes the master to perform unnecessary resource-intensive checksum calculations for each such file resource.
+
+Static catalogs avoid these problems by including metadata that refers to a specific version of the resource's file. This prevents the a newer version from being incorrectly applied, and avoids having the agent request the metadata on each Puppet run.
+
+These are the traits that make this type of catalog "static": it contains all of the information that an agent needs to determine whether the node's configuration matches the instructions and **static** state of file resources **at the point in time when the catalog was compiled.**
+
+### Static catalog features
+
+Puppet 4.4 and Puppet Server 2.3.0 enable static catalog generation by default, whether you upgrade Puppet or perform a clean installation.
```

Comment:
This is a little tricky. The `static_catalogs` setting is enabled by default, but the overall behavior is disabled by default, since users need to provide puppetserver `code-*-command` settings and scripts.